cmd/go-clang-gen/main.go

```diff
@@ -203,6 +203,50 @@ func prepareStructMembers(s *gen.Struct) {
 			}
 		}
 	}
+
+	prepareStructMembersArrayStruct(s)
+}
+
+// prepareStructMembersArrayStruct checks if the struct has two member variables, one is an array and the other a plain int/uint with size/length/count/len as its name because then this should be an array struct, and we connect them to handle a slice.
+func prepareStructMembersArrayStruct(s *gen.Struct) {
+	if len(s.Members) != 2 {
+		return
+	}
+
+	if !arrayLengthCombination(&s.Members[0].Type, &s.Members[1].Type) && !arrayLengthCombination(&s.Members[1].Type, &s.Members[0].Type) {
+		return
+	}
+
+	// If one of the members is already marked as array/slice another heuristic has already covered both members.
+	if s.Members[0].Type.IsArray || s.Members[1].Type.IsArray ||
+		s.Members[0].Type.IsSlice || s.Members[1].Type.IsSlice {
+		return
+	}
+
+	var a *gen.StructMember
+	var c *gen.StructMember
+
+	if s.Members[0].Type.PointerLevel == 1 {
```

Comment:
I do not like all of the heuristic code either. It looks hacky. Especially "headerfile.go" is a nightmare...